include/aspect/mesh_refinement/isosurfaces.h

```diff
@@ -0,0 +1,130 @@
+/*
+  Copyright (C) 2014 - 2019 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+
+#ifndef _aspect_isosurfaces_h
+#define _aspect_isosurfaces_h
+
+#include <aspect/mesh_refinement/interface.h>
+#include <aspect/simulator_access.h>
+#include <aspect/utilities.h>
+#include <deal.II/base/parsed_function.h>
+
+namespace aspect
+{
+  namespace MeshRefinement
+  {
+    namespace Internal
+    {
+      enum class PropertyName
+      {
+        Temperature,
+        Composition,
+        Background
+      };
+
+      class Property
+      {
+        public:
+          /**
+           * constructor. Make a property from a vector string and string.
+           */
+          Property(const std::string &property_name,const std::vector<std::string> &available_compositions);
+
+          PropertyName name;
+          unsigned int index;
+      };
+
+      class Isosurface
+      {
+        public:
+          /**
+           * Checks whether the provided values are between the min and max value  for
+           * those property of the isosurface. This function assumes that the order of the
+           * provided values matches the order in which the properties are stored.
+           */
+          bool values_are_in_range(const std::vector<double> values) const;
+
+          std::vector<double> min_values;
+          std::vector<double> max_values;
+          double min_refinement;
+          double max_refinement;
+          std::vector<Property> properties;
+      };
+
+      unsigned int min_max_string_to_int(const std::string &string, const unsigned int minimum_refinement_level, const unsigned int  maximum_refinement_level);
```

Comment:
What does this function do? Is this a detail of the plugin? Then move it into the .cc file.